test/src/helpers.cc

```diff
@@ -68,22 +73,111 @@ void check_partitions(
   for (const auto& p : partitions) {
     CHECK(!partitioner.done());
     CHECK(!unsplittable);
+    //TBD: *if* previous loop iteration next() happened to finish, following (next loop iteration) ...ok() will still return true,
+    //(as Status::Ok() is returned from top when already 'done()' on entry to next())
+    //tho suppose that partition check following (for actual data partition data) should fail, assuming the two last partition entries
+    //aren't the same - can the same partition information be returned more than once???
     CHECK(partitioner.next(&unsplittable).ok());
     auto partition = partitioner.current();
     check_subarray<T>(partition, p);
   }
 
   // Check last unsplittable
-  if (last_unsplittable) {
+  if (last_unsplittable) { //Was last partition/next() in prior expected to be unsplittable?
     CHECK(unsplittable);
   } else {
+    //TBD: but expectation here is that we did *not* pass all possible partitions in 'partitions', and hence
+    //there is one more - or is the 'unsplittable' return value an indication that that last 'next()' done
+    //above did *not* succeed... but then, how is the following one different from last on ein loop above that 
+    //it *would* succeed? altho, since .next() returns Ok() when already done(), and after having set
+    //unsplittable 'false', guess all could have been present and following will still be valid, tho
+    //not sure of intent (seems next() really *shouldn't* return ok, current may still have legal tho
+    //invalid data, nothing new is generated in that situation... - but next() does, so...)
     CHECK(!unsplittable);
     CHECK(partitioner.next(&unsplittable).ok());
     CHECK(!unsplittable);
     CHECK(partitioner.done());
   }
 }
 
+#if DEVING_SUBARRAY_PARTITIONER
```

Comment:
I'm going to assume this is temporary so I will ignore everything encapsulated in these flags

Reply:
Yes, temporary, appropriate to ignore for story5340.